### PR TITLE
Docs: Remove conversation ID content, update rel status name

### PR DESCRIPTION
**Pull request by jhaaaa**
- Update "General Availability" status to "Production" status according to revised status names
- Remove content about conversation IDs
- Update links
- Updated guidance about contentFallback text - now required

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,32 @@
 # xmtp-android
 
-![Test](https://github.com/xmtp/xmtp-android/actions/workflows/test.yml/badge.svg) ![Lint](https://github.com/xmtp/xmtp-android/actions/workflows/lint.yml/badge.svg) ![Status](https://img.shields.io/badge/Project_Status-Developer_Preview-yellow)
+![Test](https://github.com/xmtp/xmtp-android/actions/workflows/test.yml/badge.svg) ![Lint](https://github.com/xmtp/xmtp-android/actions/workflows/lint.yml/badge.svg) ![Status](https://img.shields.io/badge/Project_Status-Beta-yellow)
```
Comment (nplasterer) [inline README.md:3]: I honestly think we could move this to GA in this commit as well. I think it's ready.

**Review by nplasterer — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (nplasterer) [reply README.md:3]: O I guess thats what Beta is 😸

File: README.md [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-To learn more about XMTP and get answers to frequently asked questions, see [FAQ about XMTP](https://xmtp.org/docs/dev-concepts/faq).
+To learn more about XMTP and get answers to frequently asked questions, see the [XMTP documentation](https://xmtp.org/docs).
 
 ![x-red-sm](https://user-images.githubusercontent.com/510695/163488403-1fb37e86-c673-4b48-954e-8460ae4d4b05.png)
 
-## Example app
+## Quickstart app built with `xmtp-android`
 
-For a basic demonstration of the core concepts and capabilities of the `xmtp-android` client SDK, see the [Example app project](https://github.com/xmtp/xmtp-android/tree/main/example). This is currently a work in progress.
+Use the [XMTP Android quickstart app](https://github.com/xmtp/xmtp-android/tree/main/example) as a tool to start building an app with XMTP. This basic messaging app has an intentionally unopinionated UI to help make it easier for you to build with.
```
[the remark this answers is not in the document]
Comment (jhaaaa) [reply README.md:23]: Thank you for calling this out, @nplasterer - love it. We don't have DB docs yet - but just posted to dx pod to see if Pat and I can get that going! Then we'll definitely surface and link from all locations. Currently, we just have this very high level guidance: https://xmtp.org/docs/tutorials/performance WDYT? Perhaps we can link there in the meantime?

File: README.md (hunk 1 above)
Comment (jhaaaa) [reply README.md:3]: Ah, I love the idea of moving it to GA - TY - let's do it! After the renaming, GA = Production - so let me make that update.

Another requirement for GA is publishing the SDK reference docs. Thanks to you, the dokkaHtml docs are generating - I think we just wanted to find a way to publish them to a GitHub Page and then autoupdate whenever there is a merge to `main`?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jhaaaa) [reply README.md:17]: Added a link to the high-level guidance - I'll work on getting EVERYTHING to point there as well... until we get the full doc out. TY!

File: README.md (hunk 1 above)
Comment (jhaaaa) [reply README.md]: I updated the status to Production. I think I can merge this PR for now -- and we can work on getting the SDK reference docs up and linked in the README before we make a more formal announcement. WDYT, @nplasterer?

Comment (nplasterer) [reply README.md]: I think that sounds great! Lets merge this.

**Comment by nplasterer** [comment]
Dokka docs are live https://xmtp.github.io/xmtp-android/

File: README.md [hunk 3]
```diff
@@ -1,32 +1,26 @@
...
-
-Follow along in the [tracking issue](https://github.com/xmtp/xmtp-android/issues/1) for updates.
-
-To learn more about XMTP and get answers to frequently asked questions, see [FAQ about XMTP](https://xmtp.org/docs/dev-concepts/faq).
+To learn more about XMTP and get answers to frequently asked questions, see the [XMTP documentation](https://xmtp.org/docs).
 
 ![x-red-sm](https://user-images.githubusercontent.com/510695/163488403-1fb37e86-c673-4b48-954e-8460ae4d4b05.png)
 
-## Example app
+## Quickstart app built with `xmtp-android`
```
Comment (nplasterer) [inline README.md:15]: After talking with Pat I actually think we want to keep this as example.

Comment (jhaaaa) [reply README.md]: Thank you - done - changed it back!

**Review by tuanggo — COMMENTED** [review]
open

**Pull request merged**